src/core/Akka/Dispatch/MessageQueues/UnboundedPriorityMessageQueue.cs

```diff
@@ -14,9 +15,11 @@ namespace Akka.Dispatch.MessageQueues
     /// <summary> 
     /// Base class for a message queue that uses a priority generator for messages 
     /// </summary>
-    public class UnboundedPriorityMessageQueue : BlockingMessageQueue
+    public class UnboundedPriorityMessageQueue : BlockingMessageQueue, IUnboundedDequeBasedMessageQueueSemantics
```

Comment:
You are absolutely right. It does add some wierd semantics to the whole thing. I asked around on jvm akka channel about this. But haven't gotten any response as of yet.

The way this works now, is that messages are enqueued in the priority queue first. Then are handled by the actor, and then are stashed. Which means that messages are stashed in the order of priority as initially determined. When you unstash all, they are fed to the actor in that order as well.

This makes for some wierd behavior. Because it would seem to work when you simply stash all messages, and unstash all at a certain point. (see the spec i added as an example). But ofcourse this breaks down if you only intermittently stash and unstash a message.

Changing the prepend buffer to a priority queue wont solve this either. As that would mean the order would only be guarenteed within the context of the stashed messages, and not the mailbox in itself (the buffer of messages waiting will not be taken into account when determining priority)